### PR TITLE
Support for mamba init fish

### PR DESCRIPTION
**Pull request by dlukes**
See https://github.com/mamba-org/mamba/issues/1718 for details.

### CODE REVIEW

File: mamba/mamba/mamba_shell_init.py [hunk 1]
```diff
@@ -96,13 +108,11 @@ def shell_init(args):
             args.reverse,
         )
         for el in plan:
-            if el.get("function") == "init_sh_user":
```
Comment (dlukes) [inline mamba/mamba/mamba_shell_init.py:99]: I noticed the branches for `init_sh_user` and `init_sh_system` are the same. Moreover, a separate branch for `init_fish_user` would also be the same. So I tentatively merged them into one, under a single `if`. However, if separate branches are desirable for a reason I'm not aware of, I can keep the two existing ones and add a third separate one for `init_fish_user`.

File: mamba/setup.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 data_files = [
     ("etc/profile.d", ["mamba/shell_templates/mamba.sh"]),
+    ("etc/fish/conf.d", ["mamba/shell_templates/mamba.fish"]),
```
[the remark this answers is not in the document]
Comment (wolfv) [reply mamba/setup.py:24]: If that's where the conda fish scripts live, then I think that's fine!

File: mamba/mamba/shell_templates/mamba.fish [hunk 3]
```diff
@@ -0,0 +1,19 @@
+set -gx MAMBA_EXE (dirname $CONDA_EXE)/mamba
```
[the remark this answers is not in the document]
Comment (wolfv) [reply mamba/mamba/shell_templates/mamba.fish:1]: Yes, you can add the header from Anaconda, I think that's a nice gesture.

File: mamba/mamba/mamba_shell_init.py (hunk 1 above)
Comment (wolfv) [reply mamba/mamba/mamba_shell_init.py:99]: Looks good to me!

**Review by jonashaag — APPROVED** [review]
LGTM. I wonder if we can somehow deduplicate the `function micromamba` code with the stuff in `mamba.fish`

**Comment by dlukes** [comment]
> I wonder if we can somehow deduplicate the function micromamba code with the stuff in mamba.fish

Yeah, I thought about that too. They'd probably have to be generated from a template in a build step. Note it's not just the name that's different, one invokes just `$MAMBA_EXE`, the other one alternates between `$MAMBA_EXE` and `$CONDA_EXE`.

But then I thought -- realistically, these are unlikely to change often, so probably not worth the added complexity, and better keep it simple.

Let me know if you think otherwise though.

**Comment by wolfv** [comment]
Yeah, I think it's not worth the effort right now :) Thanks for making the PR! Now we just need the same for Powershell (#1088)

**Pull request merged**